serial_protocol.md

```diff
@@ -0,0 +1,72 @@
+#Protocol:
+During the first use, or when the chiplet changes environments a “NETWORK” call is expected to initialize the wifi parameters.
+
+Every time a serial connection is established we will expect a “INIT” call after which any subsequent calls can be made, until the connection is closed. 
+
+In the following examples we use $ to represent variables. For example $Host represents your firebase host.
+
+##Response Format Byte
+All responses will be prefixed with one of the following bytes signifying the response type.
+```
+  + If response is ok and a raw value
+  * If response is ok and a raw value prefixed by count of bytes in response. 
```

Comment:
I was thinking about that as well. The only use case that doesn't support is multiline strings, we cannot know if the value is a multiline string without reading the whole value. But I would like to keep the protocol able to work without the chiplet having to read the whole value into memory. It would be more flexible if it were able to stream data directly from the http connection to the serial connection. It is very possible that the chiplet will have far less ram than the client's MCU.